### PR TITLE
rkt: create parent cgroup to enable stats

### LINKED ISSUE #2400: Add stats to rkt driver
Nomad v0.5.4

The rkt driver doesn't support stat collection/reporting.  I initially thought it'd be possible to just report on the cpu, memory, etc. used by the executor, but since rkt handles container isolation and sets up cgroups for the pod, I don't think that will work.

rkt itself doesn't seem to expose any stats. but the rkt api does expose the cgroup.  So:

1. add rkt [api client](https://github.com/coreos/rkt/tree/master/api/v1alpha) to nomad
2. capture uuid of spawned pod with `--uuid-file-save`
3. call the api method `InspectPod`, which will return the cgroup, like `"/machine.slice/machine-rkt\\x2d5922fb6f\\x2db4a9\\x2d4408\\x2daf94\\x2d419a4a6efbfe.scope"`
4. use something like [github.com/crosbymichael/cgroups#Stats](https://godoc.org/github.com/crosbymichael/cgroups#Stats) to get the stats for the cgroup

I know very little about cgroups.  On one of my instances:

```
$ ls -d /sys/fs/cgroup/*"/machine.slice/machine-rkt\\x2d5922fb6f\\x2db4a9\\x2d4408\\x2daf94\\x2d419a4a6efbfe.scope"
/sys/fs/cgroup/blkio/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/cpuacct/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/cpu,cpuacct/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/cpu/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/cpuset/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/devices/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/freezer/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/hugetlb/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/memory/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/net_cls/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/net_cls,net_prio/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/net_prio/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/perf_event/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/pids/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
/sys/fs/cgroup/systemd/machine.slice/machine-rkt\x2d5922fb6f\x2db4a9\x2d4408\x2daf94\x2d419a4a6efbfe.scope
```

Does this seem like a reasonable approach?

### PR DESCRIPTION
**Pull request by schmichael**
Fixes #2400

Having the Nomad executor create parent cgroups that rkt is launched
within allows the stats collection code used for the exec driver to Just
Work. The only downside is that now the Nomad executor's resource
utilization counts against the cgroups resource limits just as it does
for the exec driver.

### CODE REVIEW

File: client/driver/rkt.go [hunk 1]
```diff
@@ -194,7 +194,10 @@ func rktGetDriverNetwork(uuid string, driverConfigPortMap map[string]string, log
 					return nil, lastErr
 				}
 
-				// This is a successful landing.
+				// This is a successful landing; log if its not the first attempt.
```
Comment (dadgar) [inline client/driver/rkt.go]: Is there a test we can add to ensure stats work?

**Comment by schmichael** [comment]
Added a test and modernized rkt tests a bit more. Also removed the env var required to run them as we already require Linux + root + rkt installed. Since it's an officially supported driver we probably shouldn't skip the tests by default.

File: client/testutil/driver_compatible.go [hunk 2]
```diff
@@ -38,13 +39,24 @@ func QemuCompatible(t *testing.T) {
 	}
 }
 
+var rktExists bool
+var rktOnce sync.Once
+
 func RktCompatible(t *testing.T) {
 	if runtime.GOOS == "windows" || syscall.Geteuid() != 0 {
-		t.Skip("Must be root on non-windows environments to run test")
+		t.Skip("Must be root on non-Windows environments to run test")
```
Comment (dadgar) [inline client/testutil/driver_compatible.go]: This would also try on Darwin

File: client/testutil/driver_compatible.go [hunk 3]
```diff
@@ -38,13 +39,24 @@ func QemuCompatible(t *testing.T) {
...
 	}
+
 	// else see if rkt exists
-	_, err := exec.Command("rkt", "version").CombinedOutput()
-	if err != nil {
+	rktOnce.Do(func() {
+		out, err := exec.Command("rkt", "version").CombinedOutput()
+		if err == nil {
+			t.Logf("found rkt version: %s", out)
```
Comment (dadgar) [inline client/testutil/driver_compatible.go]: Can we remove this log?

**Pull request merged**